src/bt_peer_connection.cpp

```diff
@@ -3545,15 +3520,19 @@ namespace libtorrent
 	}
 
 #if !defined(TORRENT_DISABLE_ENCRYPTION) && !defined(TORRENT_DISABLE_EXTENSIONS)
-	int bt_peer_connection::hit_send_barrier(std::vector<boost::asio::mutable_buffer>& iovec)
+	std::tuple<int, std::vector<boost::asio::const_buffer>>
+	bt_peer_connection::hit_send_barrier(
+		aux::array_view<boost::asio::mutable_buffer> iovec)
 	{
-		int next_barrier = m_enc_handler.encrypt(iovec);
+		int next_barrier;
+		std::vector<boost::asio::const_buffer> out_iovec;
+		std::tie(next_barrier, out_iovec) = m_enc_handler.encrypt(iovec);
 #ifndef TORRENT_DISABLE_LOGGING
 		if (next_barrier != 0)
 			peer_log(peer_log_alert::outgoing, "SEND_BARRIER"
 				, "encrypted block s = %d", next_barrier);
 #endif
-		return next_barrier;
+		return { next_barrier, out_iovec };
```

Comment:
pass on the out_iovec to the caller. Actually, I meant to say std::move() here. I will fix that.